client/tests/integration/non_mintable.rs

```diff
@@ -0,0 +1,59 @@
+#![allow(clippy::restriction)]
+
+use std::str::FromStr as _;
+
+use eyre::Result;
+use iroha_client::client;
+use iroha_data_model::{metadata::UnlimitedMetadata, prelude::*};
+use test_network::{Peer as TestPeer, *};
+
+#[test]
+fn non_mintable_asset_can_be_minted_once_but_not_twice() -> Result<()> {
+    let (_rt, _peer, mut test_client) = <TestPeer>::start_test_with_runtime();
+    wait_for_genesis_committed(&vec![test_client.clone()], 0);
+
+    // Given
+    let account_id = AccountId::from_str("alice@wonderland").expect("Valid");
+    let asset_definition_id = AssetDefinitionId::from_str("xor#wonderland").expect("Valid");
+    let create_asset = RegisterBox::new(
+        AssetDefinition::quantity(asset_definition_id.clone())
+            .mintable_once()
+            .build(),
+    );
+
+    let metadata = UnlimitedMetadata::default();
+
+    let mint = MintBox::new(
+        Value::U32(200_u32),
+        IdBox::AssetId(AssetId::new(
+            asset_definition_id.clone(),
+            account_id.clone(),
+        )),
+    );
+
+    let instructions: [Instruction; 2] = [create_asset.into(), mint.clone().into()];
+    let tx = test_client.build_transaction(instructions.into(), metadata)?;
+
+    // We can register and mint the non-mintable token
+    test_client.submit_transaction(tx)?;
+    test_client.poll_request(client::asset::by_account_id(account_id.clone()), |result| {
+        result.iter().any(|asset| {
+            asset.id().definition_id == asset_definition_id
+                && *asset.value() == AssetValue::Quantity(200_u32)
+        })
+    })?;
+
+    // We can submit the request to mint again.
+    test_client.submit_all([mint.into()])?;
+
+    // However, this will fail
+    assert!(test_client
+        .poll_request(client::asset::by_account_id(account_id), |result| {
+            result.iter().any(|asset| {
+                asset.id().definition_id == asset_definition_id
+                    && *asset.value() == AssetValue::Quantity(400_u32)
+            })
+        })
+        .is_err());
```

Comment:
Just a suggestion,
one of more explicit ways than waiting timeout would be to `submit_blocking` and query the rejected transaction and assert the rejection reason